pkg/hive/hive.go

```diff
@@ -77,20 +77,29 @@ func (s *Service) BroadcastPeers(ctx context.Context, addressee swarm.Address, p
 	return nil
 }
 
-func (s *Service) sendPeers(ctx context.Context, peer swarm.Address, peers []discovery.BroadcastRecord) error {
+func (s *Service) AddPeerAddedHandler(h func(addr swarm.Address)) {
+	s.peerHandlers = append(s.peerHandlers, h)
```

Comment:
since there's only one handler right now, i think that `peerHandlers` can be just a `func(swarm.Address)` rather than a collection, then just set the value of the pluggable function after constructing the instance

Reply:
Correct, I did it like this for libp2p as I don't expect much handlers, but here I just put it to illustrate the idiom. I can switch back to single one, and we can add more if needed later

Reply:
let's have one handler for now :+1: